File: src/Scripting/Ruby/flyweight.rb
TextStyle = Data.define(:font, :size, :color)

class StyleFactory
  def initialize
    @styles = {}
  end

  def get(font, size, color)
    key = [font, size, color]
    @styles[key] ||= TextStyle.new(font, size, color).freeze
  end

  def count
    @styles.size
  end
end

factory = StyleFactory.new
red1 = factory.get('Inter', 12, 'red')
red2 = factory.get('Inter', 12, 'red')
factory.get('Inter', 12, 'blue')
puts "styles=#{factory.count};shared=#{red1.equal?(red2)};text=ABC"

File: src/Scripting/PHP/flyweight.php
<?php
final class TextStyle { public function __construct(public string $font, public int $size, public string $color) {} }
final class StyleFactory {
    private array $styles = [];
    public function get(string $font,int $size,string $color): TextStyle { $key="$font|$size|$color"; return $this->styles[$key] ??= new TextStyle($font,$size,$color); }
    public function count(): int { return count($this->styles); }
}
$f=new StyleFactory(); $r1=$f->get('Inter',12,'red'); $r2=$f->get('Inter',12,'red'); $f->get('Inter',12,'blue');
echo 'styles='.$f->count().';shared='.(($r1===$r2)?'true':'false').';text=ABC'.PHP_EOL;
